raft-tech/TANF-app
# build(1489): Automate setting of default environment variables in Cloud.gov

## Summary of Changes

This PR eliminates the `set-backend-env-vars.sh` script and incorporates it into the `deploy-backend.sh` script as a env function. Additionally the function has been updated and adds a new set of env vars that were consistently available to Docker but not in the deployed environments. As it stands, this branch deploys successfully and is backwards compatible with the current state of `develop`.

Pull request closes #1489

## How to Test

1. To test this PR, deploy this branch to an environment and ensure the proper env vars are set at the correct time.

## Tabletop


## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] All cloud.gov applications (both tanf-dev and tanf-staging) have a consistent set of variables between backends. 
- [x] When used by the `deploy-backend.sh` or `deploy-frontend.sh`, these scripts perform as expected.
- [x] Essential environment variables are included by default in these automated scripts and do not have to manually be added by running these scripts.
- [x] Environment variables should persist across application rebuilds/restages.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
